src/mmdata.h

```diff
@@ -276,29 +277,36 @@ struct nullPntrStruct {
      */
     long poolLoc;
     /*! 
-     * allocated size of the memory block containing the \a pntrString.
+     * allocated size of the memory block containing the \a pntrString,
+     * excluding any hidden administrative data.
      * Note: this is the number of bytes, not elements!  Fixed to the size of a
      * single void* instance. 
      */
     long allocSize;
     /*! 
-     * currently used size of the memory block containing the \a pntrString.
+     * currently used size of the memory block containing the \a pntrString,
+     * excluding any hidden administrative data.
      * Note: this is the number of bytes, not elements!  Fixed to the size of a
      * single pointer element.
      */
     long actualSize;
     /*! 
      * memory for a single void* instance, set and fixed to the null pointer.
+     * A null usually marks the end of the array.
      */
     pntrString nullElement; };
 /*!
  * \var g_PntrNull. Global instance of a memory block structured like a
  * \a pntrString, but fixed in size and containing always exactly one null
  * pointer element.
  * 
- * \attention mark as const
+ * This structure must not be modified.
```

Comment:
Can we not mark it as const?

Reply:
The address of the nullElement is assigned to non-const PntrString. This leads to compiler warnings like

/home/wolf/metamath/src/doxy/src/mmdata.h:311:43: note: in expansion of macro ‘NULL_PNTRSTRING’
  311 | #define pntrString_def(x) pntrString *x = NULL_PNTRSTRING
      |                                           ^~~~~~~~~~~~~~~
/home/wolf/metamath/src/doxy/src/mmwtex.c:5102:3: note: in expansion of macro ‘pntrString_def’
 5102 |   pntrString_def(pntrTmp);
      |   ^~~~~~~~~~~~~~

I did not want to decide whether this can be resolved, so for documentation purposes I chose the weaker formulation.

Reply:
One option would be to make `g_PntrNull` const but have a cast the const away in `NULL_PNTRSTRING`. That has the "advantage" of making modification to `g_PntrNull` UB instead of just breaking everything (which makes it marginally more likely to get a compiler warning or get caught in ubsan or something).

Reply:
My suggestion is not to remove the "TODO: mark as const" in this case if you haven't conclusively determined that it is impossible.